framework/db/BaseActiveRecord.php

```diff
@@ -1762,8 +1762,8 @@ private function setRelationDependencies($name, $relation, $viaRelationName = nu
      */
     private function isAttributeDirty($attribute, $value)
     {
-        $old_attribute = $this->oldAttributes[$attribute];
-        if (is_array($value) && is_array($this->oldAttributes[$attribute])) {
+        $old_attribute = $this->_oldAttributes[$attribute];
+        if (is_array($value) && is_array($this->_oldAttributes[$attribute])) {
```

Comment:
If I read this correctly here might be a problem when `$this->_oldAttributes` is `null`.

Reply:
https://github.com/yiisoft/yii2/blob/2a7c6f2ca1b362a26ef580a280c188dda2a74510/framework/db/BaseActiveRecord.php#L642

In theory, this check ensures that this problem will never occur.